### PR TITLE
Added request tests

### PR DESCRIPTION
**Pull request by sgervais21**

### CODE REVIEW

File: request.cc [hunk 1]
```diff
@@ -20,15 +20,16 @@ bool Request::ParseRequestString(std::string request_string)
   std::string header;
   std::string status;
   std::string::size_type index;
-
+  std::string::size_type cr;
```
Comment (mjchen04) [inline request.cc]: What does this variable represent?

File: request_test.cc [hunk 2]
```diff
@@ -0,0 +1,138 @@
+#include "gtest/gtest.h"
+#include "request.h"
+#include <string>
+
+class RequestTest : public ::testing::Test {
+  protected:
+    std::string CreateRequestString(std::string method, std::string uri,
```
Comment (mjchen04) [inline request_test.cc]: Can you line up the parameters. 
Ex:
std::string method,
std::string uri,
etc

File: request_test.cc [hunk 3]
```diff
@@ -0,0 +1,138 @@
+#include "gtest/gtest.h"
+#include "request.h"
+#include <string>
+
+class RequestTest : public ::testing::Test {
+  protected:
+    std::string CreateRequestString(std::string method, std::string uri, 
+                            std::string version)
+    {
```
Comment (mjchen04) [inline request_test.cc]: Can you include the opening curly brace on the same line as the function declaration. Same thing for the other functions in this file.
Example
void func() {

}

File: request_test.cc [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+    void SetStatic(std::string stat_str, std::string file_str)
+    {
+      std::string req_str = CreateRequestString("GET", "/" + stat_str, "1.1");
+      std::map<std::string,std::string> stat_map;
+      stat_map.insert(std::make_pair(stat_str,file_str));
+      request = new Request(req_str, "echo", stat_map);
+    }
+    
+    void CleanUp() { delete request;}
```
Comment (mjchen04) [inline request_test.cc]: Add space after ; for symmetry

File: request_test.cc [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+  EXPECT_EQ(request->GetFilePath(), "src");
+  CleanUp();
+}
+
+TEST_F(RequestTest, BadStatusOnlyMethod) {
+  SetNone("GET", "", "");
+  EXPECT_FALSE(request->IsValid());
+  CleanUp();
+}
+
```
Comment (mjchen04) [inline request_test.cc:119]: Remove extra newline

File: request_test.cc [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+TEST_F(RequestTest, NoStatus) {
+  std::string req_str = "\r\n";
+  SetFromString(req_str);
+  EXPECT_FALSE(request->IsValid());
+  CleanUp();
+}
+
```
Comment (mjchen04) [inline request_test.cc]: Remove extra newline

File: static_handler_test.cc [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <fstream>
+
+class StaticHandlerTest : public ::testing::Test {
+  protected:
+    void PrepareFile(std::string directory, std::string filename, 
+                    std::string filebody)
+    {
```
Comment (mjchen04) [inline static_handler_test.cc]: Move curly brace to previous line

File: static_handler_test.cc [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+#include <string>
+#include <map>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <fstream>
+
+class StaticHandlerTest : public ::testing::Test {
+  protected:
+    void PrepareFile(std::string directory, std::string filename,
```
Comment (mjchen04) [inline static_handler_test.cc]: Line up paramaters

File: static_handler_test.cc [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+      std::string dir_str = "./" + directory;
+      if (stat(dir_str.c_str(), &st) == -1) {
+        mkdir(dir_str.c_str(), 0700);
+      }
+      std::fstream testfile;
+      testfile.open(directory + "/" + filename,std::fstream::out);
+      testfile << filebody;
+      testfile.close();
+    }
+    void HandleRequest(std::string request_string,
```
Comment (mjchen04) [inline static_handler_test.cc:28]: Line up paramaters

File: request.cc (hunk 1 above)
Comment (sgervais21) [reply request.cc]: location of carriage return

File: static_handler_test.cc [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+  EXPECT_EQ("text/html", response_->headers["Content-Type"]) << "Expected different Content-Type";
+  EXPECT_EQ(std::to_string(response_->body.size()), response_->headers["Content-Length"]) << "Expected different Content-Length";
+
+  CleanUp();
+}
+
+TEST_F(StaticHandlerTest, SimpleStaticImageRequest) {
+  std::string dir_str = "test_dir";
+  PrepareFile(dir_str, "test.jpg", 
+  "JJJJJJJJJJ");
```
Comment (mjchen04) [inline static_handler_test.cc]: Can this be combined with line 71?

**Comment by mjchen04** [comment]
Please fix the stylistic comments I added. Then you can merge.

**Pull request merged**